Jobs/Email/EditorialReminder.php

```diff
@@ -0,0 +1,185 @@
+                    $outstanding[$submissionId] = __('editor.submission.roundStatus.pendingReviewers');
+                    continue;
+                }
+
+                if ($reviewRound->getStatus() === ReviewRound::REVIEW_ROUND_STATUS_REVIEWS_COMPLETED) {
+                    $outstanding[$submissionId] = __('editor.submission.roundStatus.reviewsCompleted');
+                    continue;
+                }
+
+                if ($reviewRound->getStatus() === ReviewRound::REVIEW_ROUND_STATUS_REVIEWS_OVERDUE) {
+                    $outstanding[$submissionId] = __('editor.submission.roundStatus.reviewOverdue');
+                    continue;
+                }
+
+                if ($reviewRound->getStatus() === ReviewRound::REVIEW_ROUND_STATUS_REVISIONS_SUBMITTED) {
+                    $outstanding[$submissionId] = __('editor.submission.roundStatus.revisionsSubmitted');
+                    continue;
+                }
+            }
+
+            if (in_array($submission->getData('stageId'), [WORKFLOW_STAGE_ID_EDITING, WORKFLOW_STAGE_ID_PRODUCTION])) {
+                $lastActivityTimestamp = strtotime($submission->getData('dateLastActivity'));
+                if ($lastActivityTimestamp < strtotime('+30 days')) {
+                    /** @var WorkflowStageDAO $workflowStageDao */
+                    $workflowStageDao = DAORegistry::getDAO('WorkflowStageDAO');
+                    $outstanding[$submissionId] = __(
+                        'editor.submission.status.inactiveDaysInStage',
+                        [
+                            'days' => 30,
+                            'stage' => __($workflowStageDao->getTranslationKeyFromId($submission->getData('stageId')))
+                        ]
+                    );
+                }
+            }
+
+            if (count($outstanding) > 20) {
+                break;
+            }
+        }
+
+        if (empty($outstanding)) {
+            return;
+        }
+
+        /** @var Context $context */
+        $context = Services::get('context')->get($this->contextId);
+        $editor = Repo::user()->get($this->editorId);
+
+        // Context or user was removed since job was created, or the user was disabled
+        if (!$context || !$editor) {
+            return;
+        }
+
+        $locale = $this->getLocale($editor, $context);
+        $mailable = new MailablesEditorialReminder($context);
+        $emailTemplate = Repo::emailTemplate()->getByKey($context->getId(), $mailable::getEmailTemplateKey());
+
+        $mailable
+            ->setOutstandingTasks($outstanding, $submissions, $submissionIds->count())
+            ->from($context->getContactEmail(), $context->getName($locale))
+            ->recipients([$editor])
+            ->subject($emailTemplate->getData('subject', $locale))
+            ->body($emailTemplate->getData('body', $locale));
+
+        Mail::send($mailable);
```

Comment:
Before sending an email we must explicitly set locale for email template variables, like `$mailable->setData($locale)`. Otherwise, it will use the locale associated with the request, during which the job is getting executed if it exists or will through an error.

Reply:
Thanks @Vitaliy-1! Since this effects everything in the job, I've gone a different route and reset the active locale. Can you take a look and let me know if it's alright?

There is a bug with localizing the context name, but that's a deeper problem and I've filed an issue here: https://github.com/pkp/pkp-lib/issues/8025.

Reply:
Looks good!